cmd/kubetoken/validator.go

```diff
@@ -0,0 +1,429 @@
+	}
+	errChan := make(chan error, len(validators))
+	for _, preReqValidator := range validators {
+		wg.Add(1)
+		go func(v validator) {
+			defer wg.Done()
+			err := v.Validate(ctx)
+			if err != nil {
+				cancel()
+			}
+			errChan <- err
+		}(preReqValidator)
+	}
+
+	wg.Wait()
+	close(errChan)
+
+	for err := range errChan {
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// ctxValidator checks that the ctx use to execute the command is an okteto context
+// that has already being added to your okteto context
+type ctxValidator struct {
+	ctxName           string
+	k8sClientProvider okteto.K8sClientProvider
+}
+
+func newCtxValidator(ctxName string, k8sClientProvider okteto.K8sClientProvider) *ctxValidator {
+	return &ctxValidator{
+		ctxName:           ctxName,
+		k8sClientProvider: k8sClientProvider,
+	}
+}
+
+type errOktetoContextNotFound struct {
+	ctxName string
+}
+
+func (e errOktetoContextNotFound) Error() string {
+	return fmt.Sprintf("context '%s' not found in the okteto context store", e.ctxName)
+}
+
+type errIsNotOktetoCtx struct {
+	ctxName string
+}
+
+func (e errIsNotOktetoCtx) Error() string {
+	return fmt.Sprintf("context '%s' is not an okteto context", e.ctxName)
+}
+
+func (v *ctxValidator) Validate(ctx context.Context) error {
+	result := make(chan error, 1)
+	go func() {
+		if v.ctxName == "" {
+			result <- errEmptyContext
+			return
+		}
+		if !isURL(v.ctxName) {
+			v.ctxName = okteto.K8sContextToOktetoUrl(ctx, v.ctxName, "", v.k8sClientProvider)
+		}
+		okCtx, exists := okteto.ContextStore().Contexts[v.ctxName]
+		if !exists {
+			result <- errOktetoContextNotFound{v.ctxName}
+			return
+		}
+		if !okCtx.IsOkteto {
+			result <- errIsNotOktetoCtx{v.ctxName}
+			return
+		}
+		result <- nil
+	}()
+
+	select {
+	case err := <-result:
+		return err
+	case <-ctx.Done():
+		return ctx.Err()
+	}
+}
+
+// nsValidator validates that the namespace is valid and accessible
+type nsValidator struct {
+	ns                   string
+	nsAccessChecker      nsAccessChecker
+	previewAccessChecker previewAccessChecker
+}
+
+type errNamespaceForbidden struct {
+	ns string
+}
+
+func (e errNamespaceForbidden) Error() string {
+	return fmt.Sprintf("you don't have access to the namespace '%s'", e.ns)
+}
+
+type errPreviewForbidden struct {
+	ns string
+}
+
+func (e errPreviewForbidden) Error() string {
+	return fmt.Sprintf("you don't have access to the preview '%s'", e.ns)
+}
+
+type nsAccessChecker struct {
+	oktetoClientProvider types.OktetoClientProvider
+}
+
+func newOktetoNsAccessChecker(oktetoClientProvider types.OktetoClientProvider) nsAccessChecker {
+	return nsAccessChecker{
+		oktetoClientProvider: oktetoClientProvider,
+	}
+}
+
+func (c *nsAccessChecker) hasAccess(ctx context.Context, ns string) (bool, error) {
+	okClient, err := c.oktetoClientProvider.Provide()
+	if err != nil {
+		return false, err
+	}
+	nList, err := okClient.Namespaces().List(ctx)
```

Comment:
If we know the name of the ns I think we could try to get it instead list all of them

Reply:
prob you're right, but I don't know if this is something that the CLI must do or if it should be handled by the backend

Reply:
If the backend is the one that fails we'd probably have more logs so I'm deleting this validation